inference-engine/src/mkldnn_plugin/nodes/mkldnn_tile_node.cpp

```diff
@@ -1,45 +1,82 @@
-// Copyright (C) 2018-2020 Intel Corporation
+// Copyright (C) 2018-2021 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
 
 #include "mkldnn_tile_node.h"
-#include <legacy/ie_layers.h>
 #include <string>
 #include <mkldnn_types.h>
 #include <mkldnn_extension_utils.h>
 #include "common/cpu_memcpy.h"
+#include <ngraph/opsets/opset1.hpp>
 
 using namespace mkldnn;
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
 
-MKLDNNTileNode::MKLDNNTileNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
-        MKLDNNNode(layer, eng, cache) {}
-
-void MKLDNNTileNode::getSupportedDescriptors() {
-    auto * tileLayer = dynamic_cast<TileLayer*>(getCnnLayer().get());
+bool MKLDNNTileNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto tile = std::dynamic_pointer_cast<const ngraph::opset1::Tile>(op);
+        if (!tile) {
+            errorMessage = "Only opset1 Tile operation is supported";
+            return false;
+        }
+        if (tile->get_input_shape(TILE_INPUT).size() != tile->get_input_shape(TILE_REPEATS)[0]) {
+            errorMessage = "Doesn't support inputs with different ranks";
+            return false;
+        }
+        const auto repeatsNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(tile->get_input_node_shared_ptr(TILE_REPEATS));
+        if (repeatsNode == nullptr) {
+            errorMessage = "Only const 'repeats' input is supported";
+            return false;
+        }
+        const auto repeats = repeatsNode->cast_vector<int64_t>();
+        if (std::count_if(repeats.begin(), repeats.end(), [](int64_t x) { return x > 1; }) > 1) {
+            errorMessage = "Doesn't support 'repeats' with more than one specified axis";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
 
-    if (tileLayer == nullptr)
-        THROW_IE_EXCEPTION << "Cannot convert tile layer.";
+MKLDNNTileNode::MKLDNNTileNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
+        MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "Tile node with name '" + getName() + "'";
+
+        const auto tile = std::dynamic_pointer_cast<const ngraph::opset1::Tile>(op);
+        const auto repeatsNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(tile->get_input_node_shared_ptr(TILE_REPEATS));
+        const auto repeats = repeatsNode->cast_vector<int64_t>();
+        for (size_t i = 0; i < repeats.size(); i++) {
+            if (repeats[i] > 1) {
+                axis = i;
```

Comment:
As far as I understand uniqueness of the axis is guaranteed by ConvertTileToSeqTiles transformation. We should put the comment here about this connection, so to simplify future work for implementation generalization.

Reply:
Could you please also check that the axis and the repeats values are valid?

Reply:
done